c10/core/TensorImpl.h

```diff
@@ -138,6 +138,68 @@ struct C10_API AutogradMetaInterface {
   virtual ~AutogradMetaInterface();
 };
 
+// NOTE [ Version Counter Sharing ]
+//
+// Every Tensor has a version counter. Version counters are incremented
+// whenever the data or shape of a tensor changes through Variable operations.
```

Comment:
Do version counters update on shape change? Are you referring to `resize_`? Because if you in-place restride a tensor, I actually wouldn't expect the version counter to update (after all, the data didn't change.)

Reply:
`VariableType::as_strided_(self, size, stride, storage_offset)` actually updates the version counter of `self`. We can investigate whether this is strictly necessary, outside of this PR.

Reply:
I looked at all functions in `VariableType` that does in-place restride to a tensor (`VariableType::set_`, `VariableType::_th_set_`, `VariableType::as_strided_`), and all of them resize the tensor at the same time, which is why the version counter is bumped in those functions.

In principle we shouldn't update the version counter if we only in-place restride a tensor. I will change "shape" to "size" in this comment.

Reply:
there's no API for just restriding, so this doesn't seem relevant.  But if there was an API that let you raw inplace restride a Tensor I would absolutely expect it to update the version counter because it can change the data.